Index: src/radeon_glamor.c
===================================================================
--- src/radeon_glamor.c
+++ src/radeon_glamor.c	2019-03-08 23:36:58.645197617 +0100
@@ -238,7 +238,7 @@
 		if (info->shadow_primary) {
 			if (usage != CREATE_PIXMAP_USAGE_BACKING_PIXMAP)
 				return fbCreatePixmap(screen, w, h, depth, usage);
-		} else if (usage != CREATE_PIXMAP_USAGE_BACKING_PIXMAP) {
+		} else {
 			pixmap = glamor_create_pixmap(screen, w, h, depth, usage);
 			if (pixmap)
 			    return pixmap;
