Index: src/radeon_glamor.c
===================================================================
--- src/radeon_glamor.c
+++ src/radeon_glamor.c	2019-03-08 22:21:52.682744371 +0100
@@ -234,6 +234,16 @@
 	if (!xf86GetPixFormat(scrn, depth))
 		return NULL;
 
+    radeon_glamor_create_pixmap(ScreenPtr screen, int w, int h, int depth,
++ 		if (info->shadow_primary) {
++ 			if (usage != CREATE_PIXMAP_USAGE_BACKING_PIXMAP)
++ 				return fbCreatePixmap(screen, w, h, depth, usage);
++-		} else {
+++		} else if (usage != CREATE_PIXMAP_USAGE_BACKING_PIXMAP) {
++ 			pixmap = glamor_create_pixmap(screen, w, h, depth, usage);
++ 			if (pixmap)
++ 			    return pixmap;
+
 	if (!RADEON_CREATE_PIXMAP_SHARED(usage)) {
 		if (info->shadow_primary) {
 			if (usage != CREATE_PIXMAP_USAGE_BACKING_PIXMAP)
